exercises/concept/githup-api/.docs/instructions.md

```diff
@@ -0,0 +1,193 @@
+# Instructions
+
+The more you learn about programming, the more you start thinking that this might be more than just a fad.
+There might even be some business opportunities: people write code, they want to save it, share it, collaborate, maybe get some feedback even.
+
+Alright, let's do it, let's build a service to encourage people to code: hup, hup, people, start using git!
+Let's call it... GitHup.
+
+You get started on building a REST JSON API: users, pull requests, comments, the backend is swiftly done.
+Here is a example of the payload being served when you ask the server for the list of comments on a pull request:
+
+```json
+[
+  {
+    "id": 256,
+    "pull_request_review_id": 42,
+    "user": {
+      "id": 101,
+      "login": "octodog",
+      "avatar_url": "https://githup.com/images/error/octodog_happy.gif",
+      "site_admin": false
+    },
+    "body": "Great stuff!",
+    "side": "RIGHT",
+    "_links": {
+      "self": {
+        "href": "https://api.githup.com/repos/octodog/Hello-World/pulls/comments/1"
+      },
+      "html": {
+        "href": "https://githup.com/octodog/Hello-World/pull/1#discussion-diff-1"
+      },
+      "pull_request": {
+        "href": "https://api.githup.com/repos/octodog/Hello-World/pulls/1"
+      }
+    }
+  },
+  {
+    "id": 11,
+    "pull_request_review_id": null,
+    "user": {
+      "id": 2,
+      "login": "hexacat",
+      "name": "Alex Kate",
+      "avatar_url": "https://githup.com/images/error/hexacat_happy.gif",
+      "site_admin": true
+    },
+    "body": "Amazing stuff!",
+    "side": "LEFT",
+    "_links": {
+      "self": {
+        "href": "https://api.githup.com/repos/octodog/Hello-World/pulls/comments/1"
+      }
+    }
+  }
+]
+```
+
+Of course, the frontend will be done in Elm, but first you need to figure out what those notorious decoders/encoders are all about.
+
+```exercism/note
+Humorous story aside, the exercise requirements are taken directly from GitHub's REST API endpoints for pull request review comments.
+Some field were removed from the JSON schemas for the sake of simplicity, but this is as close to real-world applications as it gets.
+```
+
+## 1. ID, please!
+
+Your eagle eye has noticed that both comments and users have an `id` field, perfect place to start.
+
+Define `decodeId` so that it can decode a JSON object, with an integer `id`.
+
+```elm
+Decode.decodeString decodeId """{id: 10}"""
+    --> Ok 10
+```
+
+## 2. What's my name again?
+
+You notice in the example above that not all users have a name.
+
+Define `decodeName` so that it can decode a string in the `name` field if there is one.
+No name? No problem, simply return `Nothing`.
+This decoder should never fail.
```

Comment:
I don't think that's misleading, you should get `Nothing` if the name is a number.
I specify this so that they can use `Decode.maybe` like in the example `decodeMaybeFeatureId` in the about file.

Reply:
Hmmm, not sure what I was talking about here